private/pkg/manifest/module.go

```diff
@@ -35,7 +35,7 @@ var (
 
 // Blob is a blob with a digest and a content.
 type Blob interface {
-	Digest() *Digest
+	Digest() Digest
```

Comment:
This PR is just this change.

Reply:
And there's two implicit downsides here. One is Digests on the heap would be copied to stack. The other is a blob that doesn't have a digest must produce one. That latter is why chaining becomes possible: it requires blobs to have digests. Downside or upside is a matter of perspective with that one.